Automattic/wp-calypso
# Domains: Prompt to upgrade when trying to set domain as primary

#### Changes proposed in this Pull Request

This PR prompts users to upgrade when trying to set a non primary domain to primary, as part of this test: p99Zz8-10j-p2

<img width="1065" alt="Screenshot 2020-03-02 at 8 08 23 PM" src="https://user-images.githubusercontent.com/13062352/75708747-b696c880-5cc1-11ea-8408-6200bbcb3e40.png">

#### Testing instructions

For a site which has at least one registered domain:

* Verify that you get something similar to what is shown in the screenshot above when the registered domain is not primary
* Verify that no such message is shown for non-registered domains, or for domains that are primary
* Verify that clicking on Upgrade takes you to the correct page
* Ensure that this won't be shown for sites with a plan, or for users not participating in the test
* Check that everything works as expected when listing all sites under `/domains/manage`
* Make sure nothing gets broken :)
